x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -508,6 +517,11 @@ export function App({
     },
   });
 
+  const tagsIds =
+    state.persistedDoc && savedObjectsTagging
+      ? savedObjectsTagging.ui.getTagIdsFromReferences(state.persistedDoc.references)
+      : [];
```

Comment:
The hooks / effects were a little hard to follow, but accessing the references from the `lastKnownDoc` resulted in an empty array here. I had to use the persistedDoc instead.

Reply:
`lastKnownDoc` is the version of the Lens saved object passed back up from the editor component. It is pieced together here: https://github.com/elastic/kibana/blob/32abbff6858122fc86780e11af53e2e6af18370b/x-pack/plugins/lens/public/editor_frame_service/editor_frame/save.ts#L33

It doesn't have a concept of references which are not generated within the Lens editor frame itself (like those tags), that's why they get lost. One option would be to make the `getSavedObjectFormat` function aware of this type of references and keep them around, but I think it's better to keep the editor frame itself oblivious to tags.